### PR TITLE
Glossary terms v2

### PR DESCRIPTION
**Pull request by fejes713**
After a short workathon with @Chalarangelo we are finishing up Glossary terms for #690. Refer to commit history and files for terms added.

## What does your PR belong to?
- [ ] Website
- [ ] Snippets
- [x] General / Things regarding the repository (like CI Integration)
- [ ] Tests

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking improvement of a snippet)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have checked that the changes are working properly
- [x] I have checked that there isn't any PR doing the same
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: glossary/Garbage-collection.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+### Garbage collection
+
+Garbage collection is a form of automatic memory management. It attempts to reclaim memory occupied by objects that are no longer used by the program.
```
Comment (Chalarangelo) [inline glossary/Garbage-collection.md]: Split into two lines at the colon.

File: glossary/MongoDB.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+### MongoDB
+
+MongoDB stores data in flexible, JSON-like documents, meaning fields can vary from document to document and data structure can be changed over time
```
Comment (Chalarangelo) [inline glossary/MongoDB.md]: I propose that you rephrase this to:

`MongoDB is a NoSQL database model that stores data...`

File: glossary/Mutabe-value.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+### Mutable value
+
+Mutable value is a type of variable that can be changed once created.
+
+Objects are mutable as their state can be modified after they are created.
+
+Primitive values are not mutable as we perform reassignment once we change them.
```
Comment (Chalarangelo) [inline glossary/Mutabe-value.md]: Unnecessary empty lines should be removed.

File: glossary/SQL-injection.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+### SQL injection
+
+SQL injection is a code injection technique, used to attack data-driven applications.
```
Comment (Chalarangelo) [inline glossary/SQL-injection.md]: Maybe add a second line:

`SQL injections get their name from the SQL language and mainly target data stored in relational databases.`

File: glossary/SSL.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+### SSL
+
+Secure Sockets Layer known as SSL is a protocol for transferring private data across the Internet. SSL uses a cryptographic system that uses two keys to encrypt data. SSL is the predecessor to Transport Layer Security (TLS).
```
Comment (Chalarangelo) [inline glossary/SSL.md]: Split into three lines at semicolons.

**Comment by fejes713** [comment]
Thanks, @Chalarangelo - I've fixed mentioned things, hopefully, everything is ready for merging now.

File: glossary/SSL.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+### SSL
+
+Secure Sockets Layer known as SSL is a protocol for transferring private data across the Internet.
```
Comment (skatcat31) [inline glossary/SSL.md]: > Secure Socket Layer, commonly known as SSL or TLS, is a set of protocols and standards for...

Even though TLS is actually Transport Layer Security, the two terms are interchangeable since they are applied at the exact same layer in the exact same way(and if you're actually using SSL3 or below these days... I'd be surprised)

File: glossary/SSL.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
...
+SSL uses a cryptographic system that uses two keys to encrypt data.
+SSL is the predecessor to Transport Layer Security (TLS).
```
Comment (skatcat31) [inline glossary/SSL.md]: This line can be omitted since SSL and TLS are colloquially the same
